### ISSUE TITLE
Postgres-operator doesn't let to create sidecars anymore

### ISSUE THREAD

**Issue by yanchenko-igor** [zalando/postgres-operator#1206]
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** v1.5.0-69-gb379db2-dirty
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** Bare Metal K8s
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** Bug report

After upgrading to the latest image, I noticed that sidecars definition looks like this:
```
  sidecars:
  - env:
    - {}
    - {}
    - {}
    image: wrouesnel/postgres_exporter
    name: postgres-prometheus-exporter
    ports:
    - {}
    resources: {}
```
Even when I edit and insert proper values they get lost, I compared the crd definition with an older version and I see that `preserveUnknownFields: true` disappeared.

**Comment by FxKu** [comment]
We haven't set `preserveUnknownFields: true` when we were on `betav1` as it was probably the default. Seems that with `v1` we need to specify it. Although [docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning) are not super clear to me where to put it. Can you provide a PR?

**Comment by FxKu** [comment]
@yanchenko-igor can you check if this fix #1212 solves your issues?

**Comment by yanchenko-igor** [comment]
@FxKu unfortunately no, it didn't fix the issue.

**Comment by FxKu** [comment]
@yanchenko-igor ok then maybe we have to add the `x-kubernetes-preserve-unknown-fields: true` flag wherever we allow `additionalProperties`. You want to test it with sidecars section? And if this then works I create another PR.

**Comment by FxKu** [comment]
second attempt: #1228 
@yanchenko-igor

**Comment by FxKu** [comment]
I found a lot of issues in the CRD manifests which I have now fixed. I am now using `x-kubernetes-preserve-unknown-fields: true` wherever we have allowed K8s specs for our CRDs. Tested it with the sidecar in our [complete manifest](https://github.com/zalando/postgres-operator/blob/cfd83e33c8eee153960d469111cb1eeab5efa382/manifests/complete-postgres-manifest.yaml#L148) and it works now. Fields are preserved. Thanks again for raising this @yanchenko-igor !
